components/esp32/intr_alloc.c

```diff
@@ -638,21 +638,17 @@ esp_err_t esp_intr_alloc_intrstatus(int source, int flags, uint32_t intrstatusre
         if (flags&ESP_INTR_FLAG_EDGE) xthal_set_intclear(1 << intr);
         vd->source=source;
     }
-    if (flags&ESP_INTR_FLAG_IRAM) {
-        vd->flags|=VECDESC_FL_INIRAM;
-        non_iram_int_mask[cpu]&=~(1<<intr);
-    } else {
-        vd->flags&=~VECDESC_FL_INIRAM;
-        non_iram_int_mask[cpu]|=(1<<intr);
-    }
-    if (source>=0) {
-        intr_matrix_set(cpu, source, intr);
-    }
 
     //Fill return handle data.
     ret->vector_desc=vd;
     ret->shared_vector_desc=vd->shared_vec_info;
 
+    esp_intr_set_in_iram(ret, (flags & ESP_INTR_FLAG_IRAM) != 0);
```

Comment:
I could be wrong, but it looks like this refactor breaks shared interrupts - as esp_intr_set_in_iram will (silently) return ESP_ERR_INVALID_ARG and do nothing.

Reply:
yes, sorry!
ok, removed changes to `esp_intr_alloc_intrstatus`.